packages/slate/src/changes/by-key.js

```diff
@@ -422,8 +422,23 @@ Changes.replaceNodeByKey = (change, key, newNode, options = {}) => {
   const node = document.getNode(key)
   const parent = document.getParent(key)
   const index = parent.nodes.indexOf(node)
+  const { selection } = change.value
   change.removeNodeByKey(key, { normalize: false })
   change.insertNodeByKey(parent.key, index, newNode, options)
+  if (selection !== change.value.selection) {
```

Comment:
Can we express this instead as `if (anchorKey == key || focusKey == key)`? I think that would be correct still while being more clear why we're doing this. Also could you add a quick comment above this line to explain it too.

Reply:
I am not sure what you mean by `if (anchorKey == key || focusKey == key)` . 
Do you mean `if (selection.anchorKey !== change.value.selection.anchorKey || selection.focusKey !== change.value.selection.focusKey)`? 
Or Do you mean 
```
if (selection.anchorKey !== change.value.selection.anchorKey) {
  // restore anchorKey
 }
if (selection.focusKey !== change.value.selection.focusKey) {
  // restore focusKey
 }
```
If you means one of these two, I would prefer the second one~